MatrixSDK/JSONModels/Login/MXLoginSSOFlow.h

```diff
@@ -0,0 +1,37 @@
+// 
+// Copyright 2020 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+#import "MXJSONModels.h"
+#import "MXLoginSSOIdentityProvider.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+extern NSString *const MXLoginSSOFlowIdentityProvidersKey;
+
+/**
+ `MXLoginSSOFlow` represents a SSO login or a register flow supported by the home server (See MSC2858 https://github.com/matrix-org/matrix-doc/pull/2858).
+ */
+@interface MXLoginSSOFlow : MXLoginFlow
```

Comment:
I'm not very comfortable with the existence of this class and deciding that a flow is SSO or not by class check. Why don't we reuse `MXLoginFlow` for that? And maybe a method like `-(BOOL) isSSo`, which checks the type.

Reply:
I made the same as Web implementation only expose relevant properties for dedicated flow. It's not a really good idea either to expose a lot of properties that can be nil or not and used in only one case. 
In MXLoginSSOFlow case we expose the identity providers list can't be nil. And you know that is useful only in this case.

Reply:
Web implementation here https://github.com/matrix-org/matrix-react-sdk/blob/8593845e3d1120bb8d9f7f9744f4d9a3a809a578/src/Login.ts#L42

Reply:
Ok, let's continue as is then. thanks for the explanation.